core/lib/types.ts

```diff
@@ -8,20 +8,43 @@ export interface Widget<
 	Actions extends object = object,
 	Directives extends object = object
 > {
+	/**
+	 * the reactive state of the widget
+	 */
 	state$: ReadableSignal<State>;
+	/**
+	 * the different stores of the widget, all reactive
+	 */
 	stores: {[K in keyof State as `${K & string}$`]: ReadableSignal<State[K]>};
 
 	/**
 	 * Modify the parameter values, and recalculate the stores accordingly
 	 */
 	patch(parameters: Partial<Props>): void;
+	/**
+	 * the directives that can be used on associated elements to react to the widget.
```

Comment:
Directives to be used on html elements in the template of the widget itself or in the slots.